src/Dropdown/Dropdown.test.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import userEvent from '@testing-library/user-event';
+import Dropdown from './index';
+
+const mockOnToggle = jest.fn();
+
+const baseProps = {
+  show: false,
+  autoClose: true,
+  onToggle: mockOnToggle,
+};
+
+describe('<Dropdown />', () => {
+  it('renders', () => {
+    const props = {
+      ...baseProps,
+    };
+    const wrapper = mount(<Dropdown {...props} />);
+    expect(wrapper.exists()).toEqual(true);
+  });
+  it('something idk', async () => {
+    const { debug } = render(
+      <Dropdown>
+        <Dropdown.Toggle id={1}>
+          Dropdown Button
+        </Dropdown.Toggle>
+        <Dropdown.Menu>
+          <Dropdown.Item>foobar</Dropdown.Item>
+        </Dropdown.Menu>
+      </Dropdown>,
+    );
+
+    expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
+    expect(screen.queryByText('foobar')).not.toBeInTheDocument();
+
+    const button = screen.getByText('Dropdown Button');
+    userEvent.click(button);
+
+    debug();
+
+    await expect(screen.queryByText('foobar')).toBeInTheDocument();
```

Comment:
I think you want this to be await findByText instead since the findBy are the only ones that return a promise

Reply:
```
userEvent.click(document.body);
    // Why isn't the menu closing??
    debug();
    expect(screen.queryByText('foobar')).not.toBeVisible();
```
this passes but the toBeInTheDocument() fails

Reply:
try sth like 

```
expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
    expect(screen.queryByText('foobar')).not.toBeInTheDocument();
    const button = screen.getByText('Dropdown Button');
    userEvent.click(button);
    expect(screen.queryByText('foobar')).toBeInTheDocument();
    waitFor(() => expect(screen.queryByText('foobar')).toBeVisible());
    userEvent.click(document.body);
    expect(screen.queryByText('foobar')).not.toBeVisible();
```

Reply:
Very similar to @binodpant's suggestion above, this a solution I landed on that seems to work:

```jsx
screen.getByText('Dropdown Button');
expect(screen.queryByText('foobar')).not.toBeInTheDocument();
expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
expect(screen.queryByText('foobar')).not.toBeInTheDocument();

const button = screen.getByText('Dropdown Button');
userEvent.click(button);
await waitFor(() => expect(screen.queryByText('foobar')).toBeVisible());
userEvent.click(document.body);
await waitFor(() => expect(screen.queryByText('foobar')).not.toBeVisible());
```

Reply:
thanks for the feedback y'all!